packages/material-ui/src/Select/Select.test.js

```diff
@@ -237,6 +253,24 @@ describe('<Select />', () => {
       expect(React.isValidElement(selected)).to.equal(true);
     });
 
+    it('should call onChange before onClose', () => {
+      const eventLog = [];
+      const onChangeHandler = spy(() => eventLog.push('CHANGE_EVENT'));
+      const onCloseHandler = spy(() => eventLog.push('CLOSE_EVENT'));
+      const { getAllByRole, getByRole } = render(
+        <Select onChange={onChangeHandler} onClose={onCloseHandler} value="0">
+          <MenuItem value="0" />
+          <MenuItem value="1" />
+        </Select>,
+      );
+
+      fireEvent.mouseDown(getByRole('button'));
+      getAllByRole('option')[1].click();
+
+      expect(eventLog[0]).to.equal('CHANGE_EVENT');
+      expect(eventLog[1]).to.equal('CLOSE_EVENT');
```

Comment:
I think that a deep equal comparison would help the case where it fails, it would allow seeing the call order inversion more easily.

Reply:
Ok so just to be sure, you want instead of pushing strings into an array to push the entire event object and do a deep comparison or `JSON.stringify` comparison instead? I'm asking because I wasn't able to think of another way to check the order of execution of the event handlers.

I did it like this because that was the easiest way I thought of that tracks the order of which the event handlers are being called.

Reply:
I saw that you already approved but just to be sure 😅

Reply:
I was thinking of:

```jsx
      expect(eventLog).to.deep.equal(['CHANGE_EVENT', 'CLOSE_EVENT']);
```

Yeah, I have approved as I think it's already good enough to be merged.

Reply:
Ah ok, I get it. If you say so I'll update the PR, no problem.

Reply:
What do you think about it?

Reply:
We can have it like this, it looks more descriptive. I'll update the PR 😉